game/configuration.cc

```diff
@@ -102,20 +103,44 @@ namespace {
 	}
 }
 
-std::string ConfigItem::getValue() const {
-	if (m_type == "int") {
+std::string const ConfigItem::getValue() const {
+	if (this->getShortDesc() == config["audio/backend"].getShortDesc()) {
+		int AutoBackendType = 1337;
+		static int val = boost::get<int>(m_value);
+		if (val != boost::get<int>(m_value)) val = PaHostApiNameToHostApiTypeId(this->getEnumName()); // In the case of the audio backend, val is the real value while m_value is the enum case for its cosmetic name.
+		std::clog << "audio/debug: Value of selected audio backend in config.xml is: " << val << std::endl;
+		int hostApi = Pa_HostApiTypeIdToHostApiIndex(PaHostApiTypeId(val));
+		std::ostringstream oss;
+		oss << "audio/debug: Trying the selected Portaudio backend...";
+		if (val != AutoBackendType) {
+			oss << " found at index: " << hostApi;
+		}
+		else {
+			oss << " not found; but this is normal when Auto is selected."; // Auto is not a real PaHostApiTypeId, so it will always return paHostApiNotFound
+		}
+		oss << std::endl;
+		std::clog << oss.str();
+		if ((hostApi != paHostApiNotFound) || (val == AutoBackendType)) {
+		std::string backendName = (val != AutoBackendType) ? Pa_GetHostApiInfo(hostApi)->name : "Auto";
+		std::clog << "audio/info: Currently selected audio backend is: " << backendName << std::endl;
+		return backendName;
+		}
+		else std::clog << "audio/warning: Currently selected audio backend is unavailable on this system, will default to Auto." << std::endl;
+		return "Auto";
+	}
+	else if (m_type == "int") {
 		int val = boost::get<int>(m_value);
 		if (val >= 0 && val < int(m_enums.size())) return m_enums[val];
 		return numericFormat<int>(m_value, m_multiplier, m_step) + m_unit;
 	}
-	if (m_type == "float") return numericFormat<double>(m_value, m_multiplier, m_step) + m_unit;
-	if (m_type == "bool") return boost::get<bool>(m_value) ? _("Enabled") : _("Disabled");
-	if (m_type == "string") return boost::get<std::string>(m_value);
-	if (m_type == "string_list") {
+	else if (m_type == "float") return numericFormat<double>(m_value, m_multiplier, m_step) + m_unit;
```

Comment:
These `if` statements are changed to `else if`. This is not necesairy since each if returns. Please change back to if statements